### PR TITLE
Added a modal to generate contest stats for a blog post

### PR DESCRIPTION
**Pull request by sheodox**
This adds a modal with a textarea that shows some blog post html for a contest with all the generic text that'd be needed.

The modal just does a bunch of stuff with document.createElement() then returns the innerHTML of a temporary div so we should avoid XSS if it somehow managed to get in there (I haven't checked the username validation rules, but I imagine with Japanese names I've seen it is probably fairly permissive).

This required a new endpoint for admin roles, and I used the queries you sent me but the response is still kind of messy and doesn't conform to the snake case style used by other http routes. This is still pretty WIP so I don't expect this to merge as-is.

I added an editorconfig file so nvim stops using tabs for TS files. I also added a package.json script that runs prettier because until I ran that I had a lot of code that didn't conform to prettier's style.

### CODE REVIEW

File: frontend/web/package.json [hunk 1]
```diff
@@ -13,6 +13,7 @@
     "test": "jest",
     "lint": "eslint --ext .js,.ts,.tsx ./",
     "lint-fix": "eslint --fix --ext .js,.ts,.tsx ./",
+    "style-fix": "prettier --write .",
```
Comment (antonve) [inline frontend/web/package.json:16]: Suggestion: Can we name this `fmt`? That way all formatting in this project use the same command (like `go fmt`.

File: frontend/web/src/app/contest/components/modals/ViewStatsModal.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+          el.setAttribute(attr, attributes[attr])
+        }
+      }
+    }
+    return el
+  }
+
+  const winner = ranking[0]
+
+  return el('div', '', [
```
Comment (antonve) [inline frontend/web/src/app/contest/components/modals/ViewStatsModal.tsx:36]: This I would prefer to be written in idiomatic React code. Seems like it's re-implementing what React already does for us? `renderToString(element)` can be used to get the raw html from a React component.

Ref. https://reactjs.org/docs/react-dom-server.html#rendertostring

File: services/reading-contest-api/domain/contest.go [hunk 3]
```diff
@@ -43,3 +43,14 @@ type ContestID uint64
 func (id ContestID) IsGlobal() bool {
 	return id == 0
 }
+
+type ContestStats struct {
+	ByLanguage   []ContestLanguageStat `json:"byLanguage"`
+	Participants int                   `json:"participants"`
+	TotalPages   float64               `json:"totalPages"`
```
Comment (antonve) [inline services/reading-contest-api/domain/contest.go:50]: Can we call this `TotalAmount`? `Amount` is what we've been using in the code base to refer to a user's score/pages.

File: services/reading-contest-api/interfaces/repositories/contest_repository.go [hunk 4]
```diff
@@ -138,3 +138,62 @@ func (r *contestRepository) FindByID(id uint64) (domain.Contest, error) {
 
 	return contest, nil
 }
+
+func (r *contestRepository) Stats(id uint64) (domain.ContestStats, error) {
+	var contestStats domain.ContestStats
```
Comment (antonve) [inline services/reading-contest-api/interfaces/repositories/contest_repository.go:143]: For other features i'd most likely split out individual queries into different functions, but it's ok for this one. I don't see this one being changed all that often in the near future.

File: frontend/web/src/app/contest/components/modals/ViewStatsModal.tsx (hunk 2 above)
Comment (sheodox) [reply frontend/web/src/app/contest/components/modals/ViewStatsModal.tsx]: Done! Though I used `renderToStaticMarkup` which is the same thing but without any react attributes. https://reactjs.org/docs/react-dom-server.html#rendertostaticmarkup

**Review by antonve — APPROVED** [review]
Thanks!

**Pull request merged**